go/interpreter/lfvm/interpreter_test.go

```diff
@@ -183,126 +184,68 @@ func TestInterpreter_step_DetectsUpperStackLimitViolation(t *testing.T) {
-		t.Run(test.name, func(t *testing.T) {
-			ctrl := gomock.NewController(t)
-			runContext := tosca.NewMockRunContext(ctrl)
-			if test.mockCalls != nil {
-				test.mockCalls(runContext)
-			}
-			revision := tosca.R07_Istanbul
-			if test.isBerlin {
-				revision = tosca.R09_Berlin
-			}
-			if test.isLondon {
-				revision = tosca.R10_London
-			}
-			ctxt := getContext(test.code, make([]byte, 0), runContext, test.stackPtrPos, test.gasStart, revision)
-
-			// Run testing code
-			status, err := vanillaRunner{}.run(&ctxt)
-			if want, got := test.err, err; want != got {
-				t.Errorf("unexpected return: error is %v, wanted %v", got, want)
-			}
-
-			if want, got := test.endStatus, status; want != got {
-				t.Errorf("execution failed: status is %v, wanted %v", got, want)
-			}
-
-			// Check gas consumption
-			if want, got := test.gasConsumed, test.gasStart-ctxt.gas; want != got {
-				t.Errorf("execution failed: gas consumption is %v, wanted %v", got, want)
-			}
-
-			// Check gas refund
-			if want, got := test.gasRefund, ctxt.refund; want != got {
-				t.Errorf("execution failed: gas refund is %v, wanted %v", got, want)
-			}
+	for _, op := range allOpCodes() {
+		t.Run(op.String(), func(t *testing.T) {
+
+			forEachRevision(t, op, func(t *testing.T, revision tosca.Revision) {
+
+				ctrl := gomock.NewController(t)
+				mock := tosca.NewMockRunContext(ctrl)
+				// mock all to satisfy any instruction
+				mock.EXPECT().AccessAccount(gomock.Any()).Return(tosca.WarmAccess).AnyTimes()
+				mock.EXPECT().GetBalance(gomock.Any()).AnyTimes()
+				mock.EXPECT().IsAddressInAccessList(gomock.Any()).AnyTimes()
+				mock.EXPECT().GetCodeSize(gomock.Any()).AnyTimes()
+				mock.EXPECT().GetCode(gomock.Any()).AnyTimes()
+				mock.EXPECT().AccountExists(gomock.Any()).AnyTimes()
+				mock.EXPECT().AccessStorage(gomock.Any(), gomock.Any()).AnyTimes()
+				mock.EXPECT().GetStorage(gomock.Any(), gomock.Any()).AnyTimes()
+				mock.EXPECT().SetStorage(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+				mock.EXPECT().Call(gomock.Any(), gomock.Any()).AnyTimes()
+				mock.EXPECT().EmitLog(gomock.Any()).AnyTimes()
+				mock.EXPECT().SelfDestruct(gomock.Any(), gomock.Any()).AnyTimes()
+				mock.EXPECT().GetTransientStorage(gomock.Any(), gomock.Any()).AnyTimes()
+				mock.EXPECT().SetTransientStorage(gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+
+				ctx := context{
+					params: tosca.Parameters{
+						BlockParameters: tosca.BlockParameters{
+							Revision: revision,
+						},
+					},
+					context: mock,
+					// enough gas to satisfy any instruction
+					gas:    1 << 32,
+					stack:  NewStack(),
+					memory: NewMemory(),
+					code:   generateCodeFor(op),
+				}
+				err := fillStackFor(op, ctx.stack, ctx.code)
+				if err != nil {
+					t.Fatalf("unexpected creating stack: %v", err)
+				}
+
+				_, err = vanillaRunner{}.run(&ctx)
+
+				var expectedErr error
+				if !isExecutable(op) {
+					expectedErr = errInvalidOpCode
+				}
```

Comment:
A thought: since the check for the detection of invalid instructions was already covered by other tests (if I remember correctly I saw that in one of your other PRs), you could accept either an `errInvalidOpCode` or a `nil` error and avoid the need to identify invalid instructions in this test too.

Reply:
Later change deleted that test:  the reason being naming, I could not name this test dispachesAllInstructions and start filtering non-executable ones.
wdyt?